src/org/opensolaris/opengrok/util/AcceptHelper.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+ /*
+ * Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opensolaris.opengrok.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.opensolaris.opengrok.configuration.Project;
+import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
+import org.opensolaris.opengrok.history.HistoryGuru;
+import org.opensolaris.opengrok.logger.LoggerFactory;
+
+/**
+ *
+ * @author Krystof Tulinger
+ */
+public class AcceptHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AcceptHelper.class);
+
+    /**
+     * Check if I should accept this file into the index database. This method
+     * has the same effect as the call accept(null, file).
+     *
+     * @see AcceptHelper#accept(Project, File)
+     *
+     * @param file the file to check
+     * @return true if the file should be included, false otherwise
+     */
+    public static boolean accept(File file) {
+        return accept((Project) null, file);
+    }
+
+    /**
+     * Check if I should accept this file into the index database
+     *
+     * @param project if the file is relevant to some project (may be null)
+     * @param file the file to check
+     * @return true if the file should be included, false otherwise
+     */
+    public static boolean accept(Project project, File file) {
+        RuntimeEnvironment env = RuntimeEnvironment.getInstance();
+        if (!env.getIncludedNames().isEmpty()
+                && // the filter should not affect directory names
+                (!(file.isDirectory() || env.getIncludedNames().match(file)))) {
```

Comment:
I've expanded this. The question why is for @trondn because I'm quite missing the purpose of the included names here.